### PR TITLE
Implement Docker tag tracking

### PR DESCRIPTION
**Pull request by glasser**
This implements a field `dockerImage.trackMutableTag` similar to `gitConfig.trackMutableRef`.

The assumption is that when building from a particular branch or for a particular release track, every built version will be stamped with two tags: an "immutable" tag like `main---000123-abcd0123` and a "mutable" tag like `main`. The actual tag that will be put in the container spec will be the immutable tag; the automation will ensure that the tag field is updated to an immutable tag pointing at the same image that the mutable tag currently does.

This implementation assumes you are using Google's Artifact Registry and that all images that use trackMutableRef are stored in a single AR repository (ie, set of Docker image repositories).

### CODE REVIEW

File: src/update-docker-tags.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    // `main` to `main---00130-bcde` and this is deployed to staging (opening a
+    // prod promotion PR). A bug is found, so the code change X is reverted.
+    // Reproducible builds will mean that the newest tag `main---00134-dcba`
+    // will hopefully point to the same image version as `main---00123-abcd`.
+    // Choosing the min version here will mean that we will in fact "revert"
+    // staging to `main---00123-abcd`. This is now the exact same tag that is
+    // running in prod, so the prod promotion PR can auto-close rather than
+    // encouraging us to consider a no-op deploy to prod.
+    const earliestMatchingTag = min(
+      equivalentTags.filter((t) => t.startsWith(prefix)),
```
Comment (jsegaran) [inline src/update-docker-tags.ts:163]: nit: Do we need to filter again here? It looks like we do this earlier in the `equivalentTags` decleration?

File: src/update-docker-tags.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        `No tags on ${trackable.dockerImageRepository} start with '${prefix}'`,
+      );
+    }
+
+    // It's OK if the current one is null because that's what we're overwriting, but we shouldn't
+    // overwrite *to* something that doesn't exist.
+    core.info(
+      `for image ${trackable.dockerImageRepository}:${trackable.trackMutableTag}, selecting earliest label ${earliestMatchingTag}`,
+    );
+    if (trackable.tag === earliestMatchingTag) {
```
Comment (jsegaran) [inline src/update-docker-tags.ts:176]: nit: This is theoretically handled by the earlier `equivalentTags.includes(trackable.tag))`, but may be nice to keep it anyway?

**Review by jsegaran — APPROVED** [review]

File: src/update-docker-tags.ts (hunk 1 above)
Comment (glasser) [reply src/update-docker-tags.ts]: Oops, yeah, I had it here first and meant to move it rather than copy.

File: src/update-docker-tags.ts (hunk 2 above)
Comment (glasser) [reply src/update-docker-tags.ts]: Nah, happy to simplify. I didn't clean this up when I added to earlier conditional.

**Pull request merged**